### PR TITLE
[Tapioca Add-on] Fix orphaned gem RBI cleanup logic

### PR DESCRIPTION
**Pull request by alexcrocha**
### Motivation

This PR addresses an edge case where orphaned RBI files persist. 

Currently, when gem versions are bumped and then reverted, such as initially upgrading both `foo` and `bar` gems and then undoing the bump for `foo`. The lockfile diff retains changes, causing orphaned RBIs for `foo` to remain, as cleanup currently only occurs when there are no lockfile diff changes.

### Implementation

The `cleanup_orphaned_rbis` method call was moved to execute before `generate_gem_rbis`. This ensures orphaned RBIs are removed prior to generating new ones, maintaining a clean state and preventing irrelevant RBIs from lingering.

### Tests

Added a test that reproduces the specific edge case to ensure this fix effectively resolves the issue.

### CODE REVIEW

**Review by vinistock — APPROVED** [review]

File: lib/ruby_lsp/tapioca/run_gem_rbi_check.rb [hunk 1]
```diff
@@ -25,10 +25,10 @@ def initialize(project_path)
       def run
         return log_message("Not a git repository") unless git_repo?
 
+        cleanup_orphaned_rbis
```
[the remark this answers is not in the document]
Comment (vinistock) [reply lib/ruby_lsp/tapioca/run_gem_rbi_check.rb:28]: I think this is a consequence of relying on git to make these operations. They don't represent user intent to the level of context that we need.

For example, the bug you reported of bumping a gem -> generating RBIs -> downgrading the gem back. What the LSP sees is an unmodified lockfile with extraneous RBIs. How do we know if we should remove it or if these were intentional manual generation changes made by the user?

**Review by KaanOzkan — APPROVED** [review]
